plugins/TPCSDAction.cpp

```diff
@@ -48,33 +48,29 @@ namespace dd4hep {
       G4int fSpaceHitCollectionID{};
       G4int fLowPtHitCollectionID{};
       
+      G4Step* StepAtEntranceToPadRing{};
       G4ThreeVector CrossingOfPadRingCentre{};
       G4ThreeVector MomentumAtPadRingCentre{};
       G4double dEInPadRow{};
       G4double globalTimeAtPadRingCentre{};
       G4double pathLengthInPadRow{};
+
       G4double CumulativePathLength{};
       G4double CumulativeEnergyDeposit{};
+      G4double CumulativeMeanTime{};
       G4ThreeVector CumulativeMeanPosition{};
       G4ThreeVector CumulativeMeanMomentum{};
       G4int CumulativeNumSteps{};
-      
-      G4ThreeVector PreviousPostStepPosition{}; //< the end point of the previous step
-      G4int CurrentPDGEncoding{}; //< the PDG encoding of the particle causing the cumulative energy deposit
-      G4int CurrentTrackID{}; //< the TrackID of the particle causing the cumulative energy deposit
-      G4double CurrentGlobalTime{}; ///< the global time of the track causing the cumulative energy deposit
-      G4int CurrentCopyNumber{}; ///< copy number of the preStepPoint's TouchableHandle for the cumulative energy deposit
-      
+
+      G4Step* previousStep{};
```

Comment:
```suggestion
      G4Step GEANT4_CONST_STEP* previousStep{};
```
I think this would be the correct placement of the `const`, in order have the pointer modifiable, but the pointed to `G4Step` immutable.

Reply:
@tmadlener thanks!